lib/yargs-types.ts

```diff
@@ -1,17 +1,30 @@
-import { CommandInstance } from './command-types'
+import { CommandInstance } from './command'
 import { Dictionary } from './common-types'
 import { Arguments, DetailedArguments, Configuration } from 'yargs-parser'
 import { YError } from './yerror'
+import { UsageInstance } from './usage'
 
 /** Instance of the yargs module. */
 export interface YargsInstance {
   $0: string
   argv: Arguments
   customScriptName: boolean
   parsed: DetailedArguments
+  _copyDoubleDash (argv: Arguments): void
   _getLoggerInstance (): LoggerInstance
   _getParseContext(): Object
+  _hasOutput (): boolean
   _hasParseCallback (): boolean
+  // TODO: to be precised once yargs is tsified
+  _parseArgs (args: null, shortCircuit: null, _calledFromCommand: boolean, commandIndex: number): Arguments
```

Comment:
when we make the next breaking change of yargs, should we migrate any of these `_` methods to being `private` in TypeScript?

Reply:
Sure.

The next breaking change in yargs should be to switch from factories to classes, and use private and public methods to distinguish inner and API methods.

This would ease a lot understanding, IDE tools use, type inference, etc.